### DISCUSSION TITLE
Fallback mode question

### DISCUSSION THREAD

**Discussion by Nyalothas** [mswjs/msw#1246] [Q&A] [answered]
Hello 👋,

I have a react application and I integrated msw in it.
This application is integrated in another application(non-react).
When I start the app, I can use the msw and see the network requests when I'm on localhost:7700, but when I go to my application url, the msw is in fallback mode, and I can't see them.

This main application is running not on localhost, but on something like _myLocalApp.com_ served from an IIS server.
Is there an option that I could add to make the network requests be available when testing from  _myLocalApp.com_ ?

Thanks!

**Comment by kettanaito** [comment, marked as answer]
Hey, @Nyalothas. 

MSW automatically switches to a fallback mode if at least one of the following is true:

- You're using a browser that doesn't support `navigator.serviceWorker`;
- You're opening an application as a local file (`file://`).

https://github.com/mswjs/msw/blob/5378416792be4f0d573149290f920e10ffdd2c61/src/setupWorker/setupWorker.ts#L158-L159

In either of those cases it's not possible to register a Service Worker. That's why the fallback mode is here: to stub `window.fetch` and still allow you to use MSW. But since it's a stub, you won't see any requests in the Network (that's possible only when using the worker). 

Take a look at what points from the above may be the case for you.

**Reply by Nyalothas** [reply]
Thanks for the response
